packages/react-native-platform-override/src/Hash.ts

```diff
@@ -11,29 +11,52 @@ import {normalizePath, unixPath} from './PathUtils';
 import FileRepository from './FileRepository';
 import isutf8 from 'isutf8';
 
+export type HashOpts = {
+  /**
+   * What differences between files should we be insensitive to when generating
+   * a hash? Defaults to line-ending insensitivity
+   */
+  insensitivity?: 'none' | 'line-ending' | 'whitespace';
+};
+
+/**
+ * Normalize content according to insensitivty rules
+ */
+function normalizeContent(
+  content: string | Buffer,
+  opts: HashOpts,
+): string | Buffer {
+  if (
+    opts.insensitivity === 'none' ||
+    (typeof content !== 'string' && !isutf8(content))
+  ) {
+    return content;
+  }
+
+  switch (opts.insensitivity || 'line-ending') {
+    case 'line-ending':
+      // Convert to CRLF for legacy hash stability
+      return content.toString('utf8').replace(/(?<!\r)\n/g, '\r\n');
+
+    case 'whitespace':
```

Comment:
Hmm, this doesn't look right. It should keep the whitespace as this introduces hash collissions.

i.e. the strings
"abc"
"a b c"
"a        b          c"
Now all condense to the same value, where I would expect the last 2 to have the same hash, but the first different...

probabably replacing more than one whitespace to a space is the best: (\s)+ => ' '

Reply:
I saw you explicitly test that these match... So I'm okay with the behavior, but I'm still not 100% sure, maybe a comment for justification here would be nice.

Reply:
Hmm, that's probably better. Will make that change.

Reply:
The one case that's kind of interesting is something like `{foo}` vs `{ foo }` though... To do the right thing we'd probably need a real lexer though...

Normalization does seem safer than ignoring though, and is closer to what we want.